lib/src/evaluation/evaluator.dart

```diff
@@ -16,77 +16,75 @@ import 'evaluator_options.dart';
-/// After [action] completes, the evaluator is closed.
-///
-///   - [options]: The options used to configure the evaluator.
-///   - [action]: The action to perform.
-Future<T> withEvaluator<T>(EvaluatorOptions options, EvaluationAction action) async {
-  return await withEvaluatorManager((manager) async {
-    final evaluator = await manager.newEvaluator(options: options);
-    return await action(evaluator);
-  });
-}
-
-/// Like [withEvaluator] with options, but with preconfigured evaluator options.
-///
-/// - [action]: The action to perform
-Future<T> withEvaluatorPreconfigured<T>(EvaluationAction action) async {
-  return await withEvaluator(await EvaluatorOptions.preconfigured, action);
-}
-
-/// Like [withProjectEvaluator] with options, but configured with preconfigured options.
-///
-///   - [projectBaseUri]: The base path containing the PklProject file.
-///   - [action]: The action to perform.
-/// - Returns: The result of the action.
-Future<T> withProjectEvaluatorPreconfigured<T>(Uri projectBaseUri, EvaluationAction action) async {
-  return await withProjectEvaluator(projectBaseUri, await EvaluatorOptions.preconfigured, action);
-}
-
-/// Convenience method for initializing an evaluator from the project.
-///
-/// [options] is the base set of evaluator options.
-/// Any `evaluatorSettings` set within the PklProject file overwrites any fields set on [options].
-///
-/// After [action] completes, the evaluator is closed.
-/// - Parameters:
-///   - [projectBaseUri]: The base path containing the PklProject file.
-///   - [options]: The base options used to configure the evaluator.
-///   - [action]: The action to perform.
-/// - Returns: The result of the action.
-Future<dynamic> withProjectEvaluator<T>(
-  Uri projectBaseUri,
-  EvaluatorOptions options,
-  EvaluationAction action,
-) async {
-  return await withEvaluatorManager((manager) async {
-    final evaluator = await manager.newProjectEvaluator(
-      projectBaseUri: projectBaseUri,
-      options: options,
-    );
-    final result = await action(evaluator);
-    await evaluator.close();
-    return result;
-  });
-}
-
 /// The core API for evaluating Pkl modules.
+///
+/// Use the static `run` methods for one-off evaluations with automatic resource
+/// management. For managing multiple, long-lived evaluators, use an
+/// [EvaluatorManager] instance directly.
 class Evaluator {
   final int _evaluatorId;
-  final SendPort _managerSendPort; // To send commands to the background Isolate
-  final Map<int, Completer<dynamic>> _commandCompleters; // To await responses from manager
+  final SendPort _managerSendPort;
+  final Map<int, Completer<dynamic>> _commandCompleters;
+
+  // A closure provided by the manager to generate unique command IDs safely.
+  final int Function() _generateCommandId;
 
-  // Constructor only to be called by EvaluatorManager
+  /// Internal constructor. Use `Evaluator.run` or `EvaluatorManager.newEvaluator`.
   Evaluator.create({
```

Comment:
You can make a constructor private using an underscore:
```suggestion
  Evaluator._({
```

Reply:
If you pay attention, this is used outside the `Evaluator` by the `manager`. Also an intended side effect of having this public is, user can bring their own manager.  I'll annotate this as `@internal`

Reply:
> If you pay attention, this is used outside the Evaluator by the manager.

In that case, we can place the definitions of the Evaluator and the EvaluatorManager in the same file. Is there any reason we are not doing this?

> Also an intended side effect of having this public is, user can bring their own manager.

Are there use cases for bringing your own Evaluator Manager? From the other pkl implementations, I do not see such, but let me know if you have any ideas

Reply:
> In that case, we can place the definitions of the Evaluator and the EvaluatorManager in the same file. Is there any reason we are not doing this?
We could or even make it part of but I don't see this as compelling reason enough to get rid of that separation of concern. We have `@internal` annotation for cases like this. 

> Are there use cases for bringing your own Evaluator Manager? From the other pkl implementations, I do not see such, but let me know if you have any ideas
That's why I said it's unintended and also we can add features others don't have. 

I'm afraid we might be bikeshedding here. Unless you have a very strong reason to make any change here.

Reply:
> We could or even make it part of but I don't see this as compelling reason enough to get rid of that separation of concern. We have @internal annotation for cases like this.
In that case, can we just make this a default constructor?

Reply:
> > We could or even make it part of but I don't see this as compelling reason enough to get rid of that separation of concern. We have @internal annotation for cases like this.
> 
> In that case, can we just make this a default constructor (i.e unnamed)?

Done!